packages/strapi-hook-bookshelf/lib/relations.js

```diff
@@ -35,6 +35,148 @@ module.exports = {
+                `${strapiModel.collectionName}.${strapiModel.primaryKey}`,
+              );
+
+              innerJoin(
+                qb,
+                relationTable,
+                `${association.tableCollectionName}.${strapiModel.attributes[key].attribute}_${strapiModel.attributes[key].column}`,
+                `${relationTable}.${model.primaryKey}`,
+              );
+            } else {
+              const externalKey = association.type === 'collection'
+                ? `${relationTable}.${association.via}`
+                : `${relationTable}.${model.primaryKey}`;
+
+              const internalKey = association.type === 'collection'
+                ? `${strapiModel.collectionName}.${strapiModel.primaryKey}`
+                : `${strapiModel.collectionName}.${association.alias}`;
+
+              innerJoin(qb, relationTable, externalKey, internalKey);
+            }
+
+            generatePopulation(model, { relations: value });
+          } else if (_.isPlainObject(value)) {
+            generatePopulation(strapiModel, { relations: value });
+          }
+        } else {
+          generatePopulation(strapiModel, { relations: value });
+        }
+      });
+    };
+
+    return generatePopulation;
+  },
+
+  generateMatchStage: function (qb) {
+    return (strapiModel, filters) => {
+      if (!filters) {
+        return undefined;
+      }
+
+      // 1st level deep filter
+      if (filters.where) {
+        this.generateMatchStage(qb)(strapiModel, { relations: filters.where });
+      }
+
+      // 2nd+ level deep filter
+      _.forEach(filters.relations, (value, key) => {
+        if (key !== 'relations') {
+          const { associations,  attributes } = strapiModel;
+
+          // Check if the field represents an association
+          const association = associations.find(a => a.alias === key);
+          const isAnAssociationFilter = value.symbol === '=' && _.isPlainObject(value.value);
+          if (!association || !isAnAssociationFilter) {
+            const fieldKey = (association && association.nature === 'manyToMany')
+              ? `${association.tableCollectionName}.${attributes[key].attribute}_${attributes[key].column}`
+              : `${strapiModel.collectionName}.${key}`;
+            if (_.isArray(value.value)  && value.symbol !== 'IN') {
+              for (let value in value.value) {
+                if (typeof value === 'string') {
+                  value = {
+                    value,
+                    symbol: '='
+                  };
+                }
+                qb[value ? 'where' : 'orWhere'](fieldKey, value.symbol, value.value[value]);
+              }
+            } else {
+              if (!_.isObject(value)) {
+                value = {
+                  value,
+                  symbol: value.symbol || '='
+                };
+              }
+              qb.where(fieldKey, value.symbol, value.value);
```

Comment:
Question: Should we ignore the field if it doesn't exist in the model's attributes (because the actual behaviour now is that you get a 500 Internal Error message)

Reply:
@lauriejim ?

Reply:
@kamalbennani @lauriejim realistically I think it should give a 4xx error. The error is with the user and their query. Ignoring it could make it look like there is nothing wrong. It should still toss an error, but in this case it isn't the server's fault :stuck_out_tongue_winking_eye: